.pytool/Plugin/ImageValidation/ImageValidation.py

```diff
@@ -10,26 +10,164 @@
 from pathlib import Path
 from pefile import PE
 from edk2toolext.environment.plugintypes.uefi_build_plugin import IUefiBuildPlugin
-from edk2toolext.image_validation import *
-from edk2toollib.uefi.edk2.path_utilities import Edk2Path
-from edk2toollib.uefi.edk2.parsers.inf_parser import InfParser
+from edk2toolext.image_validation import (
+    Result, TestManager, TestInterface, TestWriteExecuteFlags,
+    TestSectionAlignment, MACHINE_TYPE
+)
 from edk2toollib.uefi.edk2.parsers.fdf_parser import FdfParser
 from edk2toollib.uefi.edk2.parsers.dsc_parser import DscParser
-from edk2toollib.uefi.edk2.parsers.dsc_parser import *
 import json
 from typing import List
 import logging
 from datetime import datetime
 
 
+DEFAULT_CONFIG = {
+    "TARGET_ARCH": {
+        "X64": "IMAGE_FILE_MACHINE_AMD64",
+        "IA32": "IMAGE_FILE_MACHINE_I386",
+        "AARCH64": "IMAGE_FILE_MACHINE_ARM64",
+        "ARM": "IMAGE_FILE_MACHINE_ARM"
+    },
+    "IGNORE_LIST": [],
+    "IMAGE_FILE_MACHINE_AMD64": {
+        "DEFAULT": {
+            "IMAGE_BASE": 0,
+            "DATA_CODE_SEPARATION": True,
+            "ALIGNMENT": [{
+                "COMPARISON": "==",
+                "VALUE": 4096
+            }]
+        },
+        # These are the supported MODULE_TYPEs for the X64 architecture, which use the DEFAULT profile.
+        # If a MODULE_TYPE is not listed here, but is used in the build, this check will fail.
+        "APPLICATION": {},
```

Comment:
Do we need USER_DEFINED in here? Presumably this won't run on HOST_APPLICATION bins.

Reply:
Added USER_DEFINED. Let me check the HOST_APPLICATION question. It might actually.

Reply:
Not an issue